consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/messagewrappers/Proposal.java

```diff
@@ -55,7 +73,15 @@ public static Proposal decode(final BytesValue data) {
     final SignedData<ProposalPayload> payload = SignedData.readSignedProposalPayloadFrom(rlpIn);
     final Block proposedBlock =
         Block.readFrom(rlpIn, IbftBlockHashing::calculateDataHashForCommittedSeal);
+
+    RoundChangeCertificate roundChangeCertificate = null;
```

Comment:
This is a bit nasty, be nicer if reading the round change certificate was split out into a seperate function then you could quite easily return Optional<RoundChangeCertificate> without the need for the variable initialiser.

Reply:
done.